CONTRIBUTING.rst

```diff
@@ -256,24 +263,68 @@ When developing, you can make Python changes available in R by running (from mlf
 Please also follow the recommendations from the
 `Advanced R - Style Guide <http://adv-r.had.co.nz/Style.html>`_ regarding naming and styling.
 
+Java
+~~~~
+
+If contributing to MLflow's Java APIs or modifying Java documentation,
+install `Java <https://www.java.com/>`_ and `Apache Maven <https://maven.apache.org/download.cgi>`_.
+
+Certain MLflow modules are implemented in Java, under the ``mlflow/java/`` directory.
+These are the Java Tracking API client (``mlflow/java/client``) and the Model Scoring Server
+for Java-based models like MLeap (``mlflow/java/scoring``).
+
+Other Java functionality (like artifact storage) depends on the Python package, so first install
+the Python package in a conda environment as described above.
+`Install <https://www.oracle.com/technetwork/java/javase/downloads/index.html>`_
+the Java 8 JDK (or above), and `download <https://maven.apache.org/download.cgi>`_
+and `install <https://maven.apache.org/install.html>`_ Maven. You can then build and run tests via:
+
+.. code-block:: bash
+
+  cd mlflow/java
+  mvn compile test
+
+If opening a PR that makes API changes, please regenerate API documentation as described in
+`Writing Docs`_ and commit the updated docs to your PR branch.
+
 Python
 ~~~~~~
+If you are contributing in Python, make sure that you have satisfied all the `General prerequisites and dependencies`_,
+including installing ``pytest``, as you will need it for the sections described below.
+
+Writing Python Tests
```

Comment:
Can the section related to writing and running python test come after the Python Client and Python Server? Makes a little more sense to me but open to suggestions!

Reply:
As they apply to both client and server, we decided to put them before.